internal/exec/stages/files/units.go

```diff
@@ -64,10 +119,69 @@ func (s *stage) createUnits(config types.Config) error {
 			s.relabel(relabelpath)
 		}
 	}
-	// and relabel the preset file itself if we enabled/disabled something
-	if enabledOneUnit {
-		s.relabel(util.PresetPath)
+	// if we have presets then create the systemd preset file.
+	if len(presets) != 0 {
+		if err := s.createSystemdPresetFile(presets); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// parseInstanceUnit is a helper function to extract the name
+// and coresponding instances for a given instantiated unit.
+func parseInstanceUnit(unit types.Unit) (string, string) {
+	at := strings.Index(unit.Name, "@")
+	dot := strings.Index(unit.Name, ".")
+	instance := unit.Name[at+1 : dot]
```

Comment:
I'm quite confident this will panic if the service name contains a `.`. This function should probably return an additional `error` instead.

Reply:
I'm thinking about including the following conditions for putting check around `.`
```
dot := strings.Index(unit.Name, ".")
	if dot == -1 || dot != strings.LastIndex(unit.Name, ".") {
		return "", "", err
	}
```

Reply:
Mhh, my understanding is that a dot is valid in a unit name.
At an high level, you want to:
 * find the last dot (ensure there is one), and throw away the suffix after it
 * find the @ (ensure there is one) as a separator between name and instance
 * track name and instance (ensure both are not empty)
 * return both (or return an error at any point earlier if one invariant is not met)